cpp/include/raft/neighbors/ivf_pq_types.hpp

```diff
@@ -268,69 +331,44 @@ struct index : ann::index {
         codebook_gen codebook_kind,
         uint32_t n_lists,
         uint32_t dim,
-        uint32_t pq_bits          = 8,
-        uint32_t pq_dim           = 0,
-        uint32_t n_nonempty_lists = 0)
+        uint32_t pq_bits = 8,
+        uint32_t pq_dim  = 0)
     : ann::index(),
       metric_(metric),
       codebook_kind_(codebook_kind),
-      n_lists_(n_lists),
       dim_(dim),
       pq_bits_(pq_bits),
       pq_dim_(pq_dim == 0 ? calculate_pq_dim(dim) : pq_dim),
-      n_nonempty_lists_(n_nonempty_lists),
       pq_centers_{make_device_mdarray<float>(handle, make_pq_centers_extents())},
-      pq_dataset_{make_device_mdarray<uint8_t>(handle, make_pq_dataset_extents(0))},
-      indices_{make_device_mdarray<IdxT>(handle, make_extents<IdxT>(0))},
+      lists_{make_host_mdarray<std::shared_ptr<list_data<IdxT>>>(make_extents<uint32_t>(n_lists))},
```

Comment:
I'm using `host_vector<std::shared_ptr<list_data<IdxT>>, uint32_t, row_major> lists_` at the moment. Are you looking at the latest commits? :)

Reply:
Ah, ok, why not! Fixed.

Reply:
I agree with @tfeher here. Instead of the host vector of shared pointers, I would prefer that we just use std::vector in these cases. Even though a std::vector could be used under the hood anyways, the host_vector is supposed to be treated as a host counterpart to the device_vector- thus used for storing bytes of data.